pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/files/SendFile.ios.kt

```diff
@@ -1,9 +1,83 @@
 package com.pubnub.api.endpoints.files
 
+import cocoapods.PubNubSwift.PubNubDataContentObjC
+import cocoapods.PubNubSwift.PubNubFileContentObjC
+import cocoapods.PubNubSwift.PubNubInputStreamContentObjC
+import cocoapods.PubNubSwift.PubNubObjC
+import cocoapods.PubNubSwift.PubNubUploadableObjC
 import com.pubnub.api.models.consumer.files.PNFileUploadResult
+import com.pubnub.api.v2.callbacks.Consumer
+import com.pubnub.api.v2.callbacks.Result
 import com.pubnub.kmp.PNFuture
+import kotlinx.cinterop.ExperimentalForeignApi
+import cocoapods.PubNubSwift.sendFileWithChannel
+import com.pubnub.api.PubNubException
+import com.pubnub.api.models.consumer.files.PNBaseFile
+import com.pubnub.kmp.DataUploadContent
+import com.pubnub.kmp.FileUploadContent
+import com.pubnub.kmp.StreamUploadContent
+import com.pubnub.kmp.Uploadable
+import com.pubnub.kmp.onFailureHandler
+import com.pubnub.kmp.onSuccessHandler2
+import platform.Foundation.NSInputStream
+import platform.Foundation.NSNumber
 
 /**
  * @see [PubNub.sendFile]
  */
-actual interface SendFile : PNFuture<PNFileUploadResult>
\ No newline at end of file
+actual interface SendFile : PNFuture<PNFileUploadResult>
+
+@OptIn(ExperimentalForeignApi::class)
+class SendFileImpl(
+    private val pubnub: PubNubObjC,
+    private val channel: String,
+    private val fileName: String,
+    private val inputStream: Uploadable,
+    private val message: Any?,
+    private val meta: Any?,
+    private val ttl: Int?,
+    private val shouldStore: Boolean?,
+): SendFile {
+    override fun async(callback: Consumer<Result<PNFileUploadResult>>) {
+        mapContent(inputStream)?.let { content ->
+            pubnub.sendFileWithChannel(
+                channel = channel,
+                fileName = fileName,
+                message = message,
+                content = content,
+                meta = meta,
+                ttl = ttl?.let { NSNumber(it) },
+                shouldStore = shouldStore?.let { NSNumber(it) },
+                onSuccess = callback.onSuccessHandler2 { file, timetoken ->
```

Comment:
FYI
There is file here and in line 55. I seems that they are two different object. Maybe we can rename one or the other or it doesn't make sense?